compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -50,13 +68,28 @@ static LogicalResult applyBufferizeToAllocation(RewriterBase &rewriter,
 static FailureOr<SmallVector<Value>> getOperandsFromDefOp(
     linalg::LinalgOp &linalgOp) {
   SmallVector<Value> operands;
-  for (auto input : linalgOp.getDpsInputs()) {
-    auto defOp = input.getDefiningOp();
-    // The defining op has to be a pack op, fail otherwise.
-    if (!defOp || !isa<tensor::PackOp>(defOp)) {
+  // For matmul only dispatch, we only want to fetch the input operand of the
+  // pack ops.
+  auto candidateOperands = isMatmulElementwiseFusion(linalgOp)
+                               ? linalgOp->getOperands()
+                               : linalgOp.getDpsInputs();
+  for (auto operand : candidateOperands) {
+    auto defOp = operand.getDefiningOp();
+    if (!defOp) {
       return failure();
     }
-    // We only want to fetch the input operand of the pack op.
+    // For matmul-elementwise ops fusion, there is no need to promote the
+    // operand which is the output of its producer contraction op.
+    if (isMatmulElementwiseFusion(linalgOp)) {
```

Comment:
This kind of if-else tracking is very special casing and not very robust... What are we trying to promote here?